Lib/test/test_capi/test_getargs.py

```diff
@@ -844,6 +847,8 @@ def test_y_hash(self):
         self.assertRaises(TypeError, getargs_y_hash, bytearray(b'bytearray'))
         self.assertRaises(TypeError, getargs_y_hash, memoryview(b'memoryview'))
         self.assertRaises(TypeError, getargs_y_hash, None)
+        # TypeError: must be read-only bytes-like object, not memoryview
+        self.assertRaises(TypeError, getargs_y_hash, NONCONTIG_WRITABLE)
```

Comment:
Since "y#" only works with read-only bytes-like objects, add also a test for readonly non-contiguous buffer (the same as NONCONTIG_WRITABLE, but without bytearray()).